tests/wamr-test-suites/test_wamr.sh

```diff
@@ -469,6 +472,43 @@ function wasi_test()
     echo "Finish wasi tests"
 }
 
+function wasi_certification_test()
+{
+    echo  "Now start wasi tests"
+    
+    cd ${WORK_DIR}
+    if [ ! -d "wasi-testsuite" ]; then
+        echo "wasi not exist, clone it from github"
+        git clone -b prod/testsuite-base \
+            --single-branch https://github.com/WebAssembly/wasi-testsuite.git
+    else
+        echo "wasi exists, pulling updates from github"
+        pushd wasi-testsuite
+        git fetch
+        git reset --hard origin/prod/testsuite-base
+        popd
+    fi
+
+    cd wasi-testsuite
+    python3 -m venv wasi-env && source wasi-env/bin/activate
+    python3 -m pip install -r test-runner/requirements.txt
+    IWASM_PATH=$(dirname ${IWASM_CMD})
+    PATH=${PATH}:${IWASM_PATH} python3 test-runner/wasi_test_runner.py \
+                -r adapters/wasm-micro-runtime.sh \
+                -t \
+                    tests/c/testsuite/ \
+                    tests/assemblyscript/testsuite/ \
+                | tee -a ${REPORT_DIR}/wasi_test_report.txt
+    exit_code=${PIPESTATUS[0]}
+    deactivate
```

Comment:
what's that for?

Reply:
to stop using the `wasi-env` virtual environment created before; if another test suite has dependencies it will create another virtual environment with its own dependencies

Reply:
Ah right, thanks.